src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -33,50 +33,44 @@ default boolean isSameStateAs(ReadOnlyPerson other) {
     }
 
     /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    default String getPrintableString(Printable... printables) {
+        StringBuilder builder = new StringBuilder();
+        for (Printable printable : printables) {
+            builder.append(printable.getPrintableString() + " ");
+        }
+        return builder.toString();
+    }
+
+    /**
      * Formats the person as text, showing all contact details.
      */
     default String getAsTextShowAll() {
-        final StringBuilder builder = new StringBuilder();
-        final String detailIsPrivate = "(private) ";
-        builder.append(getName())
-                .append(" Phone: ");
-        if (getPhone().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getPhone())
-                .append(" Email: ");
-        if (getEmail().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getEmail())
-                .append(" Address: ");
-        if (getAddress().isPrivate()) {
-            builder.append(detailIsPrivate);
-        }
-        builder.append(getAddress())
-                .append(" Tags: ");
+        final StringBuilder tagsBuilder = new StringBuilder();
+        tagsBuilder.append("Tags: ");
         for (Tag tag : getTags()) {
-            builder.append(tag);
+            tagsBuilder.append(tag);
         }
-        return builder.toString();
+        return getPrintableString(getName(), getPhone(), getEmail(), getAddress()) + tagsBuilder.toString();
     }
 
     /**
      * Formats a person as text, showing only non-private contact details.
      */
     default String getAsTextHidePrivate() {
         final StringBuilder builder = new StringBuilder();
```

Comment:
Everything looks good. StringJoiner is another possible alternative to StringBuiilder. Do checkout!